keras_cv/layers/preprocessing/base_image_augmentation_layer.py

```diff
@@ -293,33 +292,61 @@ def _augment(self, inputs):
         return result
 
     def _batch_augment(self, inputs):
-        return self._map_fn(self._augment, inputs)
+        augmented_batch = self._map_fn(self._augment, inputs)
+        return augmented_batch
 
     def _format_inputs(self, inputs):
+        metadata = {"is_dict": False, "use_targets": False}
         if tf.is_tensor(inputs):
             # single image input tensor
-            return {IMAGES: inputs}, False, False
+            metadata = {"is_dict": False, "use_targets": False}
+            inputs = {IMAGES: inputs}
+            return inputs, metadata
         elif isinstance(inputs, dict) and TARGETS in inputs:
             # TODO(scottzhu): Check if it only contains the valid keys
             inputs[LABELS] = inputs[TARGETS]
             del inputs[TARGETS]
-            return inputs, True, True
+            metadata = {"is_dict": True, "use_targets": True}
         elif isinstance(inputs, dict):
-            return inputs, True, False
+            metadata = {"is_dict": True, "use_targets": False}
         else:
             raise ValueError(
                 f"Expect the inputs to be image tensor or dict. Got {inputs}"
             )
+        if BOUNDING_BOXES in inputs:
+            if isinstance(inputs[BOUNDING_BOXES], tf.RaggedTensor):
+                metadata["is_bounding_boxes_ragged"] = True
+                metadata["ragged_row_lengths"] = inputs[
+                    BOUNDING_BOXES
+                ].nested_row_lengths()
+                inputs[BOUNDING_BOXES] = bounding_box.pad_with_sentinels(
+                    inputs[BOUNDING_BOXES]
+                )
+            else:
+                metadata["is_bounding_boxes_ragged"] = False
+            if tf.shape(inputs[BOUNDING_BOXES])[-1] <5:
```

Comment:
lets add this above the other checks

Reply:
I am still keeping it since it reduces computations in filtering. If we drop the padded boxes before hand, filtering will only take care of the boxes which were not dropped because of other bounding boxes being dropped during augmentataions.

Reply:
Done!